test/user-test.js

```diff
@@ -0,0 +1,158 @@
+const chai = require('chai');
+const expect = chai.expect;
+
+const User = require ('../src/user.js');
+const Recipe = require ('../src/recipe.js')
+
+describe('User', () => {
+  let userData = {}
+  let recipe; 
+  let recipe2;
+    
+  beforeEach( () => {
+    userData = {
+      "name": "Matt",
+      "id": 363,
+      "pantry": ["rice", "sugar", "salt"],
+      "favoriteRecipe": [],
+      "recipesToCook": []
+    };
+    recipe = new Recipe(595736, "https://spoonacular.com/recipeImages/595736-556x370.jpg", [
+      {
+        "id": 20081,
+        "quantity": {
+          "amount": 1.5,
+          "unit": "c"
+        }
+      },
+      {
+        "id": 18372,
+        "quantity": {
+          "amount": 0.5,
+          "unit": "tsp"
+        }
+      }], [
+      {
+        "instruction": "In a large mixing bowl, whisk together the dry ingredients (flour, pudding mix, soda and salt). Set aside.In a large mixing bowl of a stand mixer, cream butter for 30 seconds. Gradually add granulated sugar and brown sugar and cream until light and fluffy.",
+        "number": 1
+      },
+      {
+        "instruction": "Add egg and vanilla and mix until combined.",
+        "number": 2
+      }], "Loaded Chocolate Chip Pudding Cookie Cups", [
+      "antipasti",
+      "starter",
+      "snack",
+      "appetizer",
+      "antipasto",
+      "hor d'oeuvre"
+    ]);
+      
+    recipe2 = new Recipe (678353, "https://spoonacular.com/recipeImages/678353-556x370.jpg", [{
+      "id": 1009016,
+      "quantity": {
+        "amount": 1.5,
+        "unit": "cups"
+      }
+    },
+    {
+      "id": 9003,
+      "quantity": {
+        "amount": 2,
+        "unit": ""
+      }
+    }], [
+      {
+        "instruction": "Season the pork chops with salt and pepper and grill or pan fry over medium high heat until cooked, about 3-5 minutes per side. (If grilling, baste the chops in the maple dijon apple cider sauce as you grill.)Meanwhile, mix the remaining ingredients except the apple slices, bring to a simmer and cook until the sauce thickens, about 2-5 minutes.Grill or saute the apple slices until just tender but still crisp.Toss the pork chops and apple slices in the maple dijon apple cider sauce and enjoy!",
+        "number": 1
+      }], "Maple Dijon Apple Cider Grilled Pork Chops", [
+      "lunch",
+      "main course",
+      "main dish",
+      "dinner"
+    ]);
+  });
+
+  it('should be an instance of User', () => {
+    const user = new User(userData)
+    expect(user).to.be.an.instanceOf(User);
+  });
+
+  it('should contain a name', () => {
+    const user = new User(userData)
+    expect(user.name).to.equal('Matt');
+  });
+
+  it('should contain an id', () => {
+    const user = new User(userData)
+    expect(user.id).to.equal(363);
+  });
+
+  it('should contain a pantry', () => {
+    const user = new User(userData)
+    expect(user.pantry).to.deep.equal(["rice", "sugar", "salt"]);
```

Comment:
I think we want the pantry to have ingredient names and quantities so the data in this class is the same as the data in the user.js file.

Reply:
We may also want to consider is want to store the ingredient names or the ids, since all of the recipes only refer to ingredient IDs. If we go this route, we need make a lot of a calls to recipe.generateIngredientName but if we use the ids, we can compare the ids in the pantry and the ids in the recipe without needing the extra step of looking up the ingredient names.